lib/src/main/kotlin/com/swmansion/starknet/service/http/HttpRequest.kt

```diff
@@ -21,3 +23,26 @@ class HttpRequest<T>(
         return service.sendAsync(payload).thenApplyAsync(deserializer)
     }
 }
+
+class BatchHttpRequest<T>(
+    val payload: HttpService.Payload,
+    val deserializer: HttpResponseDeserializer<T>,
+    val service: HttpService,
+) : Request<List<T>> {
+    private fun parseResponse(response: HttpResponse): List<T> {
+        val responseList = Json.parseToJsonElement(response.body).jsonArray
+        return responseList.map { responseJson ->
+            val deserializedValue = deserializer.apply(HttpResponse(response.isSuccessful, response.code, responseJson.toString()))
```

Comment:
1. We should probably refactor this logic to not create fake `HttpResponse`, as this way we're repeating the same checks multiple times. We should instead handle http deserialization once, and then handle each json deserialization separately.
2. As per the json-rpc [spec](https://www.jsonrpc.org/specification#batch):
> The Response objects being returned from a batch call MAY be returned in any order within the Array.
So we probably want to reorder the responses correctly based on their `id`'s

Reply:
I've added responses ordering and reorganized deserialization. Now `buildJsonBatchHttpDeserializer` is responsible for deserializing batch response.